packages/functions/src/oidc/index.ts

```diff
@@ -1,6 +1,17 @@
-export type { AwsCredentialsProviderInit } from './aws-credentials-provider';
-export { awsCredentialsProvider } from './aws-credentials-provider';
-export {
-  getVercelOidcToken,
-  getVercelOidcTokenSync,
-} from './get-vercel-oidc-token';
+/**
+ * @deprecated Use @vercel/oidc-aws-credentials-provider instead
+ * AWS authentication for Vercel Functions
+ */
+export type { AwsCredentialsProviderInit } from '@vercel/oidc-aws-credentials-provider';
```

Comment:
I think we should re-implement rather than importing from `@vercel/oidc-aws-credentials-provider`.

Re-implementing will remove the dependency on `@vercel/oidc-aws-credentials-provider` and allow it to retain optional peer dependency `@aws-sdk/credential-provider-web-identity` while allowing `@vercel/oidc-aws-credentials-provider` to have a dependency (not peer) on `@aws-sdk/credential-provider-web-identity`.

The import on `@vercel/oidc` is fine though.